test/python/transpiler/test_time_unit_conversion.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2025.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Test the optimize-1q-gate pass"""
+
+import itertools
+from ddt import ddt, idata, unpack
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Delay
+from qiskit.circuit.library import XGate
+from qiskit.transpiler import InstructionDurations, InstructionProperties, Target
+from qiskit.transpiler.passes import TimeUnitConversion
+from qiskit.converters import circuit_to_dag
+from qiskit.transpiler.exceptions import TranspilerError
+from test import QiskitTestCase  # pylint: disable=wrong-import-order
+
+
+@ddt
+class TestTimeUnitConversion(QiskitTestCase):
+    """Test for time unit conversion pass."""
+
+    def setUp(self):
+        super().setUp()
+
+        self.dt = 1e-9
+        self.x_dt = 10
+        self.delay_dt = 5
+
+        self.target = {}
+        self.target["dt"] = Target(dt=self.dt)
+        self.target["dt"].add_instruction(
+            XGate(),
+            properties={(0,): InstructionProperties(duration=self.x_dt * self.dt)},
+        )
+        self.target["nodt"] = Target()
+        self.target["nodt"].add_instruction(XGate())
+        self.target["none"] = None
+
+        self.durations = {
+            "si": InstructionDurations([("x", None, self.x_dt * self.dt, None, "s")]),
+            "dt": InstructionDurations([("x", None, self.x_dt, None, "dt")]),
+            "none": None,
+        }
+
+    @idata(itertools.product(("s", "dt"), ("dt", "si", "none"), ("dt", "nodt", "none")))
+    @unpack
+    def test_gate_and_delay(self, delay_unit, inst_durations_unit, target_type):
```

Comment:
Perhaps the name here could be a bit more specific?

Reply:
Most of the labeling was in the `TestTimeUnitConversion` and ddt-label parts of the test name, but the tests are moved into test_transpiler now in [b749029](https://github.com/Qiskit/qiskit/pull/14027/commits/b7490299ec032dc501770e0a912175a89c20a90f).